worker/dbaccessor/tracker.go

```diff
@@ -144,7 +144,9 @@ func (w *trackedDBWorker) TxnNoRetry(ctx context.Context, fn func(context.Contex
 	}
 
 	db := w.db
-	w.mutex.RUnlock()
+	// We can't release the mutex before the transaction ends; because the db can be closed in tomb loop in another
+	// goroutine and cause issues in TU TestWorkerAttemptsToVerifyDBButSucceedsWithDifferentDB (and may be a bug).
+	defer w.mutex.RUnlock()
```

Comment:
You might as well put this defer where you acquire, and delete the unlock on error.

Reply:
Yes, I was thinking of that, but I want to be sure that the missing defer was not intentional first.

(because when i don't see the usual pattern `lock;defer unlock`, I assume it may be intentional.)

I updated the code to make it cleaner.